akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -706,26 +723,18 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
     val localMembers = localGossip.members
     val localOverview = localGossip.overview
     val localSeen = localOverview.seen
-    val localReachability = localOverview.reachability
+    val localReachability = localGossip.teamReachability(selfTeam)
 
-    // check if the node to DOWN is in the `members` set
-    localMembers.find(_.address == address) match {
-      case Some(m) if (m.status != Down) ⇒
+    // check if the node to DOWN is in the team `members` set
+    localMembers.find(m ⇒ m.team == selfTeam && m.address == address) match {
```

Comment:
this is wrong, it should be allowed to down a node in another team
(but that is not the reason for the failure in SurviveNetworkInstabilitySpec)